src/main/webapp/app/app.component.ts

```diff
@@ -28,4 +28,14 @@ export class AppComponent implements OnInit {
   deleted(): void {
     this.tasks$ = this.taskService.getAll();
   }
+
+  completed(): void {
+    this.tasks$ = this.taskService.getAll();
+  }
+
+  async clearCompleted(): Promise<void> {
```

Comment:
Why the change to use Promises over continuing using Observable?

Reply:
To avoid callback hell and we need to make calls synchronously one after another.  

OfCourse we could have used flatMap or switchMap Rxjs operator to achieve the same behaviour but you don’t always need to use Observables. Promises are still here, and it’s OK to use them when you need it.